scripts/mosh.pl

```diff
@@ -396,7 +396,7 @@ sub predict_check {
     delete $ENV{ 'SSH_CONNECTION' };
     chdir; # $HOME
     print "MOSH IP ${userhost}\n";
-    exec( $server, @server );
```

Comment:
I'm wondering, is there any reason to use `shell_quote()` here at all?  We could just do `exec( "server " . join( " ", @server )`, since we're not running ssh here.

Reply:
Joining an array with spaces and no quoting is *always the wrong thing to do*. It’s a fundamentally lossy operation. In this case, it would manifest by corrupting `@command`: `mosh --local ::1 -- cmd 'arg with spaces'` would result in `cmd arg with spaces`. Even if that bug weren’t as easy to manifest as it is, it would still be locally a bug. Don’t do it. *Don’t dooo ittt.*

Reply:
Looks good to me (unsurprisingly...), though as I mentioned I didn't test it myself.

Reply:
Actually, looking at this again, this code does break if there's a separator or metacharacter in the executable pathname (such as `/Users/cgull/spacey directory/mosh/src/frontend/mosh-server`).  I believe it should be `exec( shell_quote( $server, @server ) )`, so as to quote the executable name too.  I *think* I'm understanding Perl correctly here.  @andersk, concur?

Reply:
That doesn't sound right. If the goal is to make `--server='env foo=bar mosh-server'` work with and without `--local`, then spaces in `$server` should *not* be quoted.

Reply:
Right, like @TomiBelan said, `--server` is intentionally passed without quoting so the shell can split it (19884d6c85c15b44b10c23c42d5687d1f9667bcd), and the point of this PR is to fix `--local` to match that behavior.

Reply:
I started looking at a shell quoting problem in `src/tests` that this interacts with and forgot about the original problem...

Pulled now.